tests/src/Type/data/bug-414.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Bug414;
+
+use Drupal\Core\Access\AccessResultInterface;
+use Drupal\Core\Entity\EntityInterface;
+use function PHPStan\Testing\assertType;
+
+function (EntityInterface $entity): void {
+    assertType('bool', $entity->access('view'));
+    assertType(AccessResultInterface::class, $entity->access('view', null, true));
+};
+
+function (EntityInterface $entity): void {
+    $accessArgs = ['delete'];
+    $access = $entity->access(...$accessArgs);
+    assertType('bool', $access);
+
+    $accessArgs = ['delete', null, false];
+    $access = $entity->access(...$accessArgs);
+    assertType('bool', $access);
+
+    $accessArgs = ['delete', null, true];
+    $access = $entity->access(...$accessArgs);
+    assertType(AccessResultInterface::class, $access);
```

Comment:
@dpi so, this still fails even with the new stub. I haven't dug in via xdebug, yet. It's like the parameters are not analyzed correctly when using spreads.

Reply:
Replicated on phpstan.dev: https://phpstan.org/r/7afd799d-3169-4ac9-a69b-7debbdc9d3d5